From: Ben Hutchings <ben@decadent.org.uk>
Date: Mon, 10 Sep 2012 01:09:04 +0100
Subject: ahci: Add JMicron 362 device IDs

commit 1fefb8fdc6562057a0e4e4542f3d4323981c9686 upstream.

The JMicron JMB362 controller supports AHCI only, but some revisions
use the IDE class code.  These need to be matched by device ID.

These additions have apparently been included by QNAP in their NAS
devices using these controllers.

References: http://bugs.debian.org/634180
Signed-off-by: Ben Hutchings <ben@decadent.org.uk>
Signed-off-by: Jeff Garzik <jgarzik@redhat.com>
---
 drivers/ata/ahci.c |    3 +++
 1 file changed, 3 insertions(+)

Index: linux-3.2.46/drivers/ata/ahci.c
===================================================================
--- linux-3.2.46.orig/drivers/ata/ahci.c	2013-07-26 18:54:05.000000000 +0000
+++ linux-3.2.46/drivers/ata/ahci.c	2013-07-26 19:24:57.000000000 +0000
@@ -306,6 +306,9 @@
 	/* JMicron 360/1/3/5/6, match class to avoid IDE function */
 	{ PCI_VENDOR_ID_JMICRON, PCI_ANY_ID, PCI_ANY_ID, PCI_ANY_ID,
 	  PCI_CLASS_STORAGE_SATA_AHCI, 0xffffff, board_ahci_ign_iferr },
+	/* JMicron 362B and 362C have an AHCI function with IDE class code */
+	{ PCI_VDEVICE(JMICRON, 0x2362), board_ahci_ign_iferr },
+	{ PCI_VDEVICE(JMICRON, 0x236f), board_ahci_ign_iferr },
 
 	/* ATI */
 	{ PCI_VDEVICE(ATI, 0x4380), board_ahci_sb600 }, /* ATI SB600 */
